CRM/Member/Form.php

```diff
@@ -465,4 +472,40 @@ public function testSubmit($formValues) {
     $this->submit();
   }
 
+
+  /**
+   * Build membership info array, which is used when membership type is selected.
+   *
+   * Data is used to:
+   * - set the payment information block
+   * - set the max related block
+   *
+   * @param array $values
+   * @param string $totalAmount
+   * @param string $taxAmount
+   *   (only used from membership renewal)
+   *
+   * @return array
+   */
+  protected function getMembershipInfoArray($values, $totalAmount, $taxAmount = NULL) {
+    // @todo - we really should assign the tax rate for the financial_type_id to the template.
```

Comment:
opps this todo can go ... I did & later I found using it improved the tpl - but thinking to hold this & rebase when the trait pr is merged